### PR TITLE
Optionally disallow the expansion of unfielded negations

### PR DESCRIPTION
**Pull request by ivakegg**

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FixUnfieldedTermsVisitor.java [hunk 1]
```diff
@@ -300,51 +314,17 @@ public Object visit(ASTReferenceExpression node, Object data) {
         }
     }
     
-    /**
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FixUnfieldedTermsVisitor.java]: moved to parent class (ParallelIndexExpansion)

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FixUnfieldedTermsVisitor.java [hunk 2]
```diff
@@ -353,12 +333,18 @@ protected IndexLookupCallable buildIndexLookup(JexlNode node, boolean positive)
...
     }
     
+    /**
+     * Expand if we have an unfielded identifier
+     *
+     * @param node
+     * @return true if contains an unfielded identifier
+     */
     @Override
-    protected boolean isIdentifier(JexlNode node) {
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FixUnfieldedTermsVisitor.java]: This was a very poorly named method....

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ParallelIndexExpansion.java [hunk 3]
```diff
@@ -487,24 +520,71 @@ private boolean computeExpansionForSubtree(JexlNode node, Join join, Map<JexlNod
...
+    // This flag should keep track of whether we are in a negated portion of the tree.
+    protected boolean negated = false;
+    
+    protected void negate() {
+        this.negated = !this.negated;
+    }
+    
+    protected void revertNegation() {
+        this.negated = !this.negated;
+    }
```
Comment (FineAndDandy) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ParallelIndexExpansion.java]: combine to single method toggleNegation()

Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ParallelIndexExpansion.java]: will do

**Pull request merged**